apps/condo/domains/common/components/GraphQlSearchInput.tsx

```diff
@@ -46,20 +48,23 @@ export const GraphQlSearchInput: React.FC<ISearchInputProps> = (props) => {
 
     useEffect(() => {
         handleSearch('')
-    }, [])
+    },  researchDeps ? researchDeps : [])
```

Comment:
Inconsistent react component lyfecycle logic, need to be reworked.

Reply:
```
useEffect(() => {
        handleSearch('')
}, [])
```

this will invoked on component mount

```
useEffect(() => {
        handleSearch('')
},  researchDeps)
```

Idc when this is will be invoked and updated. all deps in useEffect need's to be provided as [...deps]

```
useEffect(() => {
        handleSearch('')
},  [researchDeps]) 
```

This can add behaviour u wants to implement. But, it's looks really strange